src/main/java/com/hivemq/configuration/reader/RestrictionConfigurator.java

```diff
@@ -31,12 +36,56 @@ public RestrictionConfigurator(@NotNull final RestrictionsConfigurationService r
 
 
     void setRestrictionsConfig(@NotNull final RestrictionsEntity restrictionsEntity) {
-        restrictionsConfigurationService.setMaxConnections(restrictionsEntity.getMaxConnections());
-        restrictionsConfigurationService.setMaxClientIdLength(restrictionsEntity.getMaxClientIdLength());
-        restrictionsConfigurationService.setNoConnectIdleTimeout(restrictionsEntity.getNoConnectIdleTimeout());
-        restrictionsConfigurationService.setIncomingLimit(restrictionsEntity.getIncomingBandwidthThrottling());
-        restrictionsConfigurationService.setMaxTopicLength(restrictionsEntity.getMaxTopicLength());
+        restrictionsConfigurationService.setMaxConnections(validateMaxConnections(restrictionsEntity.getMaxConnections()));
+        restrictionsConfigurationService.setMaxClientIdLength(validateMaxClientIdLength(restrictionsEntity.getMaxClientIdLength()));
+        restrictionsConfigurationService.setNoConnectIdleTimeout(validateNoConnectIdleTimeout(restrictionsEntity.getNoConnectIdleTimeout()));
+        restrictionsConfigurationService.setIncomingLimit(validateIncomingLimit(restrictionsEntity.getIncomingBandwidthThrottling()));
+        restrictionsConfigurationService.setMaxTopicLength(validateMaxTopicLength(restrictionsEntity.getMaxTopicLength()));
+    }
+
+    private long validateMaxConnections(final long maxConnections) {
+        if (maxConnections == UNLIMITED_CONNECTIONS) {
+            return maxConnections;
+        }
+        if (maxConnections < MAX_CONNECTIONS_MINIMUM) {
+            log.warn("The configured max-connections ({}) must be greater than 0. The value was set to {} (unlimited) instead.", maxConnections, MAX_CONNECTIONS_DEFAULT);
+            return MAX_CONNECTIONS_DEFAULT;
+        }
+        return maxConnections;
+    }
+
+    private int validateMaxClientIdLength(final int maxClientIdLength) {
+        if (maxClientIdLength < MAX_CLIENT_ID_LENGTH_MINIMUM || maxClientIdLength > MAX_CLIENT_ID_LENGTH_MAXIMUM) {
+            log.warn("The configured max-clientid-length ({}) must be in the range {} - {}. The value was set to {} instead.", MAX_CLIENT_ID_LENGTH_MINIMUM, MAX_CLIENT_ID_LENGTH_MAXIMUM, maxClientIdLength, MAX_CLIENT_ID_LENGTH_DEFAULT);
+            return MAX_CLIENT_ID_LENGTH_DEFAULT;
+        }
+        return maxClientIdLength;
+    }
+
+    private int validateMaxTopicLength(final int maxTopicLength) {
+        if (maxTopicLength < MAX_TOPIC_LENGTH_MINIMUM || maxTopicLength > MAX_TOPIC_LENGTH_MAXIMUM) {
+            log.warn("The configured max-topic-length ({}) must be in the range {} - {}. The value was set to {} instead.", MAX_TOPIC_LENGTH_MINIMUM, MAX_TOPIC_LENGTH_MINIMUM, maxTopicLength, MAX_TOPIC_LENGTH_DEFAULT);
```

Comment:
```suggestion
            log.warn("The configured max-topic-length ({}) must be in the range {} - {}. The value was set to {} instead.", maxTopicLength, MAX_TOPIC_LENGTH_MINIMUM, MAX_TOPIC_LENGTH_MINIMUM, MAX_TOPIC_LENGTH_DEFAULT);
```

Reply:
You are right, I made a mistake here. But your suggestion is also incorrect. Fixed it anyway.